database/Reservation.js

```diff
@@ -38,66 +65,97 @@ var make = (booking) => {
+          //  overlap = new reservation start OR end is between existing reservation start AND end
+          if ((currResStart >= exResStart && currResStart <= exResEnd) ||
             (currResEnd >= exResStart && currResEnd <= exResEnd)) {
-          overlaps.push(reservations[i]);
-          //  Does the name match?  If so, decline "you're alread here"
-          if (booking.name !== '' && booking.name === reservations[i].customer_name) {
-            nameMatch = true;
-            break;
+            overlaps.push(reservations[i]);
+            //  Does the name match?  If so, decline "you're alread here"
+            if (booking.name !== '' && booking.name === reservations[i].customer_name) {
+              nameMatch = true;
+              break;
+            }
           }
         }
-      }
-      //  it works!
-      var notification = '';
-      if (nameMatch) {
-        notification = "Sorry, you already have a booking at this time.  Please make another selection.";
-      } else if (overlaps.length >= AVAIL_TABLES) {
-        nofitication = "Sorry, there are no bookings available on this date and time.  Please make another selection.";
-      } else {
-        //  add reservation to the database
-        notification = 'Your reservation has been successfully booked!';
-        addReservation(booking);
-      }
-
-      resolve(notification)
-    });
+        //  it works!
+        var notification = '';
+        if (nameMatch) {
+          notification = 'Sorry, you already have a booking at this time.  Please make another selection.';
+        } else if (overlaps.length >= AVAIL_TABLES) {
+          nofitication = 'Sorry, there are no bookings available on this date and time.  Please make another selection.';
+        } else {
+          //  add reservation to the database
+          notification = 'Your reservation has been successfully booked!';
+          addReservation(booking);
+        }
+
+        resolve(notification)
+      });
   });
 };
 
+//Ron's crud update: here is were I'm going to insert a put/update request to update/change a reservation
+
 addReservation = (booking) => {
-  var res = new Reservation(
-    {
-      restaurant_id: booking.restId,
-      customer_name: booking.name,
-      reservation_time: booking.time,
-      guests: booking.guests
-    });
+  var res = new Reservation({
+    reservation_id: booking.reservId,
+    restaurant_id: booking.restId,
+    customer_name: booking.name,
+    reservation_time: booking.time,
+    guests: booking.guests
+  });
 
   // save model to database
   res.save(function (err, booking) {
     if (err) return console.error(err);
   });
 };
 
+//!! RON'S ADDITION FOR THE PUT REQUEST TO UPDATE THE RESERVATION
+
+updateReservation = (err, booking) => {
+  console.log('reservation.update:', booking);
+  if (err) throw err;
+  var data = db.reservations("customer_name")
+  var reservationQuery = {
+    customer_name: 'Heath Spinka'
+  };
```

Comment:
if you are implementing a reservation_Id, it's easiest to query by that id since it's unique, where as querying by name might yield multiple reservations